docs/workflow-lifecycle.md

```diff
@@ -0,0 +1,239 @@
+# Workflow lifecycle
+
+Below we follow a typical sequence of events in the execution of the following very simple workflow pseudocode:
+
+```javascript
+function myWorkflow() {
+  result = callActivity(myActivity);
+  return result;
+}
+```
+
+<br>
+
+---
+
+_Note that below, "Initialize history", "Append event to history", or "Persist Mutable State and history tasks", all refer to durable writes to the persistence layer._
+
+<br>
+
+1. **The User Application sends a `StartWorkflowExecution` request**
+   - Workflow History is initialized with `[WorkflowExecutionStarted, WorkflowTaskScheduled]`
+   - A Workflow Task is added in the Matching service
+
+```mermaid
+sequenceDiagram
+User Application->>Frontend: StartWorkflowExecution
+Frontend->> History: StartWorkflowExecution
+History ->> Persistence: CreateWorkflowExecution
+note over Persistence: Initialize History:<br>[WorkflowExecutionStarted, WorkflowTaskScheduled]<br>persist MutableState and Transfer Task
+Persistence ->> History: Create Succeed
+History->>Frontend: Start Succeed
+Frontend->>User Application: Start Succeed
+loop QueueProcessor
+    History->>Persistence: GetHistoryTasks
+		History->>History: ProcessTask
+		History->>Matching: AddWorkflowTask<br>
+end
+```
+
+<details>
+<summary><i>Code entrypoints</i></summary>
+
+- History service [`StartWorkflow` handler](https://github.com/temporalio/temporal/blob/ef49189005b5323c532264287af6c08a447aab8a/service/history/api/startworkflow/api.go#L157).
+- History service [queue processors](https://github.com/temporalio/temporal/blob/ef49189005b5323c532264287af6c08a447aab8a/service/history/history_engine.go#L303) and [transfer task queue processor](https://github.com/temporalio/temporal/blob/ef49189005b5323c532264287af6c08a447aab8a/service/history/queues/queue_immediate.go#L150).
+
+</details>
+<br>
+
+---
+
+<br>
+
+2. **The Worker dequeues and processes the Workflow Task**
+   - It advances the workflow execution, and becomes blocked on the Activity call.
+
+```mermaid
+sequenceDiagram
+Worker->>Frontend: PollWorkflowTask
+Frontend->>Matching: PollWorkflowTask
+History->>Matching: AddWorkflowTask (step 1 above)
+Matching->>History: RecordWorkflowTaskStarted
+History->>Persistence: UpdateWorkflowExecution
+note over Persistence: append History Event: WorkflowTaskStarted<br>update Mutable State & add Timer Task (workflow task timeout)
+Persistence->>History: Update Succeed
+History->>Matching: Record Succeed
+Matching->>Frontend: WorkflowTask
+Frontend->>Persistence: GetHistoryEvents
+Persistence->>Frontend: History Events
+Frontend->>Worker: WorkflowTask
+Worker->>Worker: Advance workflow
+```
+
+<details>
+<summary><i>Code entrypoints</i></summary>
+
+- History service [`RecordWorkflowTaskStarted` handler](https://github.com/temporalio/temporal/blob/ef49189005b5323c532264287af6c08a447aab8a/service/history/handler.go#L319)
+
+</details>
+<br>
+
+---
+
+<br>
+
+3. **The Worker sends a `ScheduleActivityTask` command**
```

Comment:
A little more context here might be helpful—why would the worker send `ScheduleActivityTask`? For example:

```suggestion
3. **The workflow launches an activity, causing the Worker to send a `ScheduleActivityTask` command back to the Frontend**
```